psi-java/src/main/kotlin/io/spine/tools/psi/java/PsiCommands.kt

```diff
@@ -29,12 +29,62 @@ package io.spine.tools.psi.java
 import com.intellij.openapi.command.CommandProcessor
 import io.spine.tools.psi.java.Environment.commandProcessor
 import io.spine.tools.psi.java.Environment.project
+import kotlin.system.exitProcess
 
 /**
- * Executes the given [Runnable] as a PSI modification
- * [command][CommandProcessor.executeCommand].
+ * Executes the given [runnable] as a PSI modification [command][CommandProcessor.executeCommand],
+ * ensuring any errors thrown by the [runnable] are handled with the provided [errorHandler].
+ *
+ * By default, the `CoreCommandProcessor` used for PSI command processing suppresses
+ * any [Throwable] thrown by the command, making it difficult for PSI users to detect
+ * and handle errors.
+ *
+ * To address it, this method wraps the given [runnable] in a try-catch block.
+ * Any thrown errors or exceptions are passed to the provided [errorHandler].
+ *
+ * If no custom [errorHandler] is provided, the default behavior is to log the error
+ * to [System.err] and terminates the process with a non-zero exit code.
+ * Use [executeSilent] to suppress errors entirely.
+ *
+ * @param runnable The [Runnable] to execute as a PSI modification.
+ * @param errorHandler A lambda to handle any [Throwable] thrown by the [runnable].
+ *  Default to logging the error and terminating the process.
+ *
+ * @see executeSilent
  */
 @JvmName("execute")
-public fun execute(runnable: Runnable) {
+@Suppress("TooGenericExceptionCaught") // We need everything, including `java.lang.Error`.
+public fun execute(runnable: Runnable, errorHandler: (Throwable) -> Unit = ::logAndTerminate) {
+    val withHandledErrors = Runnable {
+        try {
+            runnable.run()
+        } catch (t: Throwable) {
+            errorHandler(t)
+        }
+    }
+    executeSilent(withHandledErrors)
```

Comment:
This code combination is confusing because it won't result in the really silent execution.